### PR TITLE
Fixes and Improvements

### PR DESCRIPTION
**Pull request by Mbucari**
- No more Invoke() or ShowDialogSynchronously(). Everything window/form related MUST be async.  This change requires the changes I made to AudibleApi.
- Added preprocessor definitions and platform logging to AppScaffolding to use for logging so we know which platform Libation is running on.
- Fixed the custom directory/subdirectory issue. #334
- Fixed publish commands for Linux and MacOS so they will always be the correct platform assemblies.  This involded adding commands to delete the bin\ and obj\ folders for all publish targets on `dotnet clean`, and then calling `clean` before publishing for each platform.

**Publish Info**

~I've figured out why the publish bins weren't working for the Windows build. For some reason, LibationCli has System.Drawing.Common v5.0 as a dependency, but the net6.0-windows apps have System.Drawing.Common v6.0 as a dependency. No matter which order you build the projects in you'll only ever end up with the v5.0 version in the built files.  I tried tracking down the source of that dependency but couldn't find it.~

~And for that reason, I removed Hangover project from the chardonnay build.  We just need to make an Avalonia Hangover, which I'll do soonish.~

**Edited** Updated to your Lucene fix and now all published packages work as intended!

**New Publish commands for chardonnay**

Windows
```console
dotnet clean   -c Release
dotnet publish -c Release -o bin\Release\win-chardonnay   LibationAvalonia\LibationAvalonia.csproj -p:PublishProfile=LibationAvalonia\Properties\PublishProfiles\WindowsProfile.pubxml
dotnet publish -c Release -o bin\Release\win-chardonnay   LibationCli\LibationCli.csproj           -p:PublishProfile=LibationCli\Properties\PublishProfiles\WindowsProfile.pubxml
dotnet publish -c Release -o bin\Release\win-chardonnay   Hangover\Hangover.csproj                 -p:PublishProfile=Hangover\Properties\PublishProfiles\WindowsProfile.pubxml
```

Linux
```console
dotnet clean   -c Release
dotnet publish -c Release -o bin\Release\linux-chardonnay LibationAvalonia\LibationAvalonia.csproj -p:PublishProfile=LibationAvalonia\Properties\PublishProfiles\LinuxProfile.pubxml
dotnet publish -c Release -o bin\Release\linux-chardonnay LibationCli\LibationCli.csproj           -p:PublishProfile=LibationCli\Properties\PublishProfiles\LinuxProfile.pubxml
```

Mac
```console
dotnet clean   -c Release
dotnet publish -c Release -o bin\Release\macos-chardonnay LibationAvalonia\LibationAvalonia.csproj -p:PublishProfile=LibationAvalonia\Properties\PublishProfiles\MacOSProfile.pubxml
dotnet publish -c Release -o bin\Release\macos-chardonnay LibationCli\LibationCli.csproj           -p:PublishProfile=LibationCli\Properties\PublishProfiles\MacOSProfile.pubxml
```

**New Publish commands for classic**

```console
dotnet clean   -c Release
dotnet publish -c Release -o bin\Release\win-classic      LibationCli\LibationCli.csproj           -p:PublishProfile=LibationCli\Properties\PublishProfiles\WindowsProfile.pubxml
dotnet publish -c Release -o bin\Release\win-classic      LibationWinForms\LibationWinForms.csproj -p:PublishProfile=LibationWinForms\Properties\PublishProfiles\WindowsProfile.pubxml
dotnet publish -c Release -o bin\Release\win-classic      Hangover\Hangover.csproj                 -p:PublishProfile=Hangover\Properties\PublishProfiles\WindowsProfile.pubxml
```

### CODE REVIEW

**Comment by rmcrackan** [comment]
> For some reason, LibationCli has System.Drawing.Common v5.0 as a dependency

This version of Lucene (our search engine) was only ever created for .net framework, not core. I created/maintain [this](https://github.com/rmcrackan/LuceneNet303r2) for compatibility. To make it work in dotnet core requires System.Configuration.ConfigurationManager which for some reason needs System.Drawing.Common. Upgrading now.

File: Source/AudibleUtilities/AudibleUtilities.csproj [hunk 1]
```diff
@@ -5,10 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AudibleApi" Version="4.5.0.1" />
-  </ItemGroup>
-
-  <ItemGroup>
+    <ProjectReference Include="..\..\..\audible api\AudibleApi\AudibleApi\AudibleApi.csproj" />
```
Comment (rmcrackan) [inline Source/AudibleUtilities/AudibleUtilities.csproj:8]: revert. also update your references. I approved your audibleapi PR and upgraded the lucene nuget to upgrade that system drawing common v5

File: Source/LibationAvalonia/Program.cs [hunk 2]
```diff
@@ -30,6 +30,14 @@ static async Task Main()
 			var classicLifetimeTask = Task.Run(() => new ClassicDesktopStyleApplicationLifetime());
 			var appBuilderTask = Task.Run(BuildAvaloniaApp);
 
+			if (App.IsWindows)
```
Comment (rmcrackan) [inline Source/LibationAvalonia/Program.cs:33]: Ah, I misunderstand that one comment of yours regarding this. I like this. Much thanks.

Comment (Mbucari) [reply Source/LibationAvalonia/Program.cs:33]: To be clear, this is one of the 2 ways that version/platform are tracked.  The GUI apps set ReleaseIdentifier, which is logged in LibationScaffolding.logStartupState().

In addition, the new preprocessor definitions in AppScaffolding.csproj are also used by LibationScaffolding.logStartupState() to log the OS.

File: Source/AudibleUtilities/AudibleUtilities.csproj (hunk 1 above)
Comment (Mbucari) [reply Source/AudibleUtilities/AudibleUtilities.csproj]: Done

Comment (Mbucari) [reply Source/AudibleUtilities/AudibleUtilities.csproj]: And no you can `publish` all 4 versions successfully!

**Comment by rmcrackan** [comment]
1) With the System.Drawing.Common v5.0 dependency fixed, is the added 'clean' step still needed in build?
2) I currently use dotnet **build** for win classic and win chardonnay; not dotnet **publish**. Which win deployment, if any, would you like for me to change to publish?

**Comment by Mbucari** [comment]
> 1. With the System.Drawing.Common v5.0 dependency fixed, is the added 'clean' step still needed in build?

Absolutely.  VS just doesn't clean up dependencies when building for a new platform.  It's the reason the guy in the Mac theread said you packed Linux bins. You didn't make a mistake, VS used the wrong assemblies.

2. I currently use dotnet **build** for win classic and win chardonnay; not dotnet **publish**. Which win deployment, if any, would you like for me to change to publish?

I updated the build commands.  With the Lucene fix we can do publish on all 4 release builds.  I updated the command list in my first post to include publish commands for win classic.

I tested on my windows VM.  Before your Lucene fix, I was getting the System.Drawing.Common.dll not found error.  After updating Lucene, everything works as intended.

**Comment by Mbucari** [comment]
Now that we got this publishing thing sorted out, I'm going to try building standalone for Linux again. It might work this time with the clean step.

**Comment by rmcrackan** [comment]
Thanks for thinking of clean. I assumed with targeting different output folders that they couldn't interfere with one another.

**Pull request merged**